packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts

```diff
@@ -435,22 +439,44 @@ export class ScheduleTrigger implements INodeType {
 				Second: moment.tz(timezone).format('ss'),
 				Timezone: moment.tz(timezone).format('z Z'),
 			};
-			const lastExecutionWeekNumber =
+			const lastExecution =
 				recurrencyRuleIndex !== undefined
 					? staticData.recurrencyRules[recurrencyRuleIndex]
 					: undefined;
 
 			// Checks if have the right week interval, handles new years as well
-			if (weeksInterval && recurrencyRuleIndex !== undefined) {
+			if (
+				intervalSize &&
+				recurrencyRuleIndex !== undefined &&
+				(typeInterval === 'weeks' || typeInterval === 'undefined')
+			) {
 				if (
-					lastExecutionWeekNumber === undefined || // First time executing this rule
-					moment.tz(timezone).week() >= weeksInterval + lastExecutionWeekNumber || // not first time, but minimum interval has passed
-					moment.tz(timezone).week() + 52 >= weeksInterval + lastExecutionWeekNumber // not first time, correct interval but year has passed
+					lastExecution === undefined || // First time executing this rule
+					moment.tz(timezone).week() >= intervalSize + lastExecution || // not first time, but minimum interval has passed
+					moment.tz(timezone).week() + 52 >= intervalSize + lastExecution // not first time, correct interval but year has passed
 				) {
 					staticData.recurrencyRules[recurrencyRuleIndex] = moment.tz(timezone).week();
 					this.emit([this.helpers.returnJsonArray([resultData])]);
 				}
 				// There is no else block here since we don't want to emit anything now
+			} else if (intervalSize && recurrencyRuleIndex !== undefined && typeInterval === 'days') {
```

Comment:
This is starting to become a bit spaghetti - maybe refactor to take into accounts scenarios based on the recurrence rule?